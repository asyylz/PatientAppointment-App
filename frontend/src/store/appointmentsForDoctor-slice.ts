import {
  fetchEntitiesWithIdAndToken,
  createEntitySlice,
} from './create-generic-slice';

export const doctorAppointmentsUrl = (id: string) => {
  return `http://localhost:3000/api/v1/appointments/doctors/${id}`;
};

export const fetchAppointmentsForDoctor =
  fetchEntitiesWithIdAndToken<AppointmentsForDoctor>(
    'appointmentsForDoctor',
    doctorAppointmentsUrl
  );

const appointmentsForDoctorSlice = createEntitySlice<AppointmentsForDoctor>(
  'appointmentsForDoctor',
  fetchAppointmentsForDoctor
);

const { actions, reducer } = appointmentsForDoctorSlice;

export const appointmentsForDoctorActions = actions;
export default reducer;
