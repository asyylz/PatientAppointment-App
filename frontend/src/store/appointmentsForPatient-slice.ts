import {
  fetchEntitiesWithIdAndToken,
  createEntitySlice,
} from './create-generic-slice';


export const patientAppointmentsUrl = (id: string, pagination?: number) => {
  return `http://localhost:3000/api/v1/appointments/patients/${id}?limit=10&page=${pagination}`;
};

export const fetchAppointmentsForPatient =
  fetchEntitiesWithIdAndToken<AppointmentsForPatient>(
    'appointmentsForPatient',
    patientAppointmentsUrl
  );

const appointmentsForPatientSlice = createEntitySlice<AppointmentsForPatient>(
  'appointmentsForPatient',
  fetchAppointmentsForPatient
);

const { actions, reducer } = appointmentsForPatientSlice;

export const appointmentsForPatientActions = actions;
export default reducer;
