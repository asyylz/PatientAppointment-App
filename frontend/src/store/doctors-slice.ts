import { fetchEntities, createEntitySlice } from './create-generic-slice';
import { PayloadAction } from '@reduxjs/toolkit';

// export const doctorsUrl = (pagination?: number, departmentId?: string) => {
//   return `http://localhost:3000/api/v1/doctors?limit=2&page=${pagination}&sort=firstName${
//     departmentId ? `&departmentId=${departmentId}` : ''
//   }`;
// };

export const doctorsUrl = (url: string) => {
  return url ;
};

// Fetch doctors thunk
export const fetchDoctors = fetchEntities<Doctor>('doctors', 'get');

// Create doctors slice
const doctorsSlice = createEntitySlice<Doctor>(
  'doctors',
  fetchDoctors,
  (builder) => {
    builder.addCase(
      'doctors/selectDoctor',
      (state: ExtendedEntityState<Doctor>, action: PayloadAction<Doctor>) => {
        state.selectedDoctor = action.payload;
        console.log('State.selectedDoctor updated:', state.selectedDoctor); // Debug state update
      }
    );
  }
);

// Export the actions and reducer
const { actions, reducer } = doctorsSlice;

export const doctorActions = {
  ...actions,
  selectDoctor: (doctor: Doctor) => ({
    type: 'doctors/selectDoctor',
    payload: doctor,
  }),
};

export default reducer;
