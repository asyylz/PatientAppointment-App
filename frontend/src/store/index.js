import { configureStore } from '@reduxjs/toolkit';

const store = configureStore({
  reducer: { departments: departmentsSlice.reducer }, // root reducer
});

export default store;
