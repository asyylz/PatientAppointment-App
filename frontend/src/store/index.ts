import { configureStore, combineReducers } from '@reduxjs/toolkit';
import {
  persistReducer,
  persistStore,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import departmentsReducer from './departments-slice';
import doctorsReducer from './doctors-slice';
import reviewsReducer from './reviews-slice';
import currentUserReducer from './currentUser-slice';
//import { useDispatch } from 'react-redux';
import searchReducer from './search-slice';
import appointmentsForDoctorReducer from './appointmentsForDoctor-slice';
import appointmentsForPatientReducer from './appointmentsForPatient-slice';
import storage from 'redux-persist/lib/storage'; // defaults to localStorage for web

// const rootPersistConfig = {
//   key: 'root',
//   storage,
// };

// const persistedReducer = persistReducer(rootPersistConfig, currentUserReducer);

// const store = configureStore({
//   reducer: {
//     currentUser: persistedReducer,
//     departments: departmentsReducer,
//     doctors: doctorsReducer,
//     reviews: reviewsReducer,
//     search: searchReducer,
//     appointmentsForDoctor: appointmentsForDoctorReducer,
//     appointmentsForPatient: appointmentsForPatientReducer,
//   },
//   middleware: (getDefaultMiddleware) =>
//     getDefaultMiddleware({
//       serializableCheck: {
//         ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
//         ignoredActionPaths: ['meta.arg', 'payload.timestamp'],
//         ignoredPaths: ['items.dates'],
//       },
//     }),
//   devTools: process.env.NODE_ENV !== 'production',
// });

// const rootPersistConfig = {
//   key: 'root',
//   storage,
//   whitelist: ['currentUser', 'doctors'], // Only persist the 'currentUserSlice' slice
// };

const authPersistConfig = {
  key: 'currentUser',
  storage: storage,
  // blacklist: ['status', 'error'],
};
const doctorPersistConfig = {
  key: 'doctors',
  storage: storage,
  // blacklist: ['status','error']
};
export const allReducers = combineReducers({
  currentUser: persistReducer(authPersistConfig, currentUserReducer),
  departments: departmentsReducer,
  doctors: persistReducer(doctorPersistConfig, doctorsReducer),
  reviews: reviewsReducer,
  search: searchReducer,
  appointmentsForDoctor: appointmentsForDoctorReducer,
  appointmentsForPatient: appointmentsForPatientReducer,
});

//const persistedReducer = persistReducer(rootPersistConfig, allReducers);

const store = configureStore({
  reducer: allReducers,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }),
  devTools: process.env.NODE_ENV !== 'production',
});

export type AppDispatch = typeof store.dispatch;
//export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export type AppStore = typeof store;
export type RootState = ReturnType<AppStore['getState']>;

export const persistor = persistStore(store);

export default store;
