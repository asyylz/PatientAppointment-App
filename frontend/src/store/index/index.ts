import { configureStore, combineReducers } from '@reduxjs/toolkit';
import {
  persistReducer,
  persistStore,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import departmentsReducer from '../departments-slice/departments-slice';
import doctorsReducer from '../doctors-slice/doctors-slice';
import reviewsReducer from './../reviews-slice/reviews-slice';
import currentUserReducer from '../currentUser-slice/currentUser-slice';
import searchReducer from '../search-slice/search-slice';
import appointmentsForDoctorReducer from '../appointmentsForDoctor-slice/appointmentsForDoctor-slice';
import appointmentsForPatientReducer from '../appointmentsForPatient-slice/appointmentsForPatient-slice';
import storage from 'redux-persist/lib/storage'; // defaults to localStorage for web

const authPersistConfig = {
  key: 'currentUser',
  storage: storage,
  // blacklist: ['status', 'error'],
};

const doctorPersistConfig = {
  key: 'doctors',
  storage: storage,
  // blacklist: ['status','error']
};

export const allReducers = combineReducers({
  currentUser: persistReducer(authPersistConfig, currentUserReducer),
  departments: departmentsReducer,
  doctors: persistReducer(doctorPersistConfig, doctorsReducer),
  reviews: reviewsReducer,
  search: searchReducer,
  appointmentsForDoctor: appointmentsForDoctorReducer,
  appointmentsForPatient: appointmentsForPatientReducer,
});

const store = configureStore({
  reducer: allReducers,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [
          FLUSH,
          REHYDRATE,
          PAUSE,
          PERSIST,
          PURGE,
          REGISTER,
          'appointmentsForPatient/fetchWithId/rejected',
          'appointmentsForDoctor/fetchWithId/rejected',
          'departments/fetch/rejected',
          'doctors/fetch/rejected',
        ],
      },
    }),
  devTools: process.env.NODE_ENV !== 'production',
});

export function setupStore(reducer?: any, preloadedState?: Partial<RootState>) {
  //console.log('from index store', preloadedState); // i can see my preloaded state here  with mock data
  const store = configureStore({
    reducer: reducer,
    preloadedState,
  });

  //console.log(store.getState()); // but still cant see current user but can see the rest of the states
  return store;
}

export type AppDispatch = typeof store.dispatch;
export type AppStore = typeof store;
export type RootState = ReturnType<AppStore['getState']>;

export const persistor = persistStore(store);

export default store;
