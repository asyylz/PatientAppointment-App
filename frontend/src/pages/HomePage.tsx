import { useEffect, useState } from 'react';
import classes from './HomePage.module.css';
import { useSelector } from 'react-redux';
import { fetchDoctors } from './../store/doctors-slice';
import { useDispatch } from 'react-redux';
import { AppDispatch } from '../store';

const images: { [key: string]: string } = {
  one: 'https://images.unsplash.com/photo-1499728603263-13726abce5fd?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8NHx8aGVhbHRofGVufDB8fDB8fHwy',
  two: 'https://images.unsplash.com/photo-1488228469209-c141f8bcd723?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MzB8fGhlYWx0aHxlbnwwfHwwfHx8Mg%3D%3D',
  three:
    'https://images.unsplash.com/photo-1444312645910-ffa973656eba?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8Nnx8aGVhbHRofGVufDB8fDB8fHwy',
  four: 'https://images.unsplash.com/photo-1530497610245-94d3c16cda28?q=80&w=1364&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D',
  five: 'https://images.unsplash.com/photo-1604881991720-f91add269bed?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8Mjc1fHxoZWFsdGh8ZW58MHx8MHx8fDI%3D',
};
export default function HomePage() {
  const dispatch: AppDispatch = useDispatch();
  const {
    entities: doctors,
    // status,
    // error,
  } = useSelector((state: RootState) => state.doctors);

  const [currentImageIndex, setCurrentImageIndex] = useState(0);

  useEffect(() => {
    dispatch(
      fetchDoctors(
        'http://localhost:3000/api/v1/doctors?limit=4&page=1&sort=firstName'
      )
    );
  }, [dispatch]);

  useEffect(() => {
    const imageKeys = Object.keys(images);
    const interval = setInterval(() => {
      setCurrentImageIndex((prevIndex) => (prevIndex + 1) % imageKeys.length);
    }, 3000);

    return () => clearInterval(interval);
  }, []);
  // console.log(token);
  return (
    <div className={classes.wrapper}>
      <div className={`${classes.container} ${classes['container--body']}`}>
      
        <section
          className={`${classes.container} ${classes['container--section1']}`}
        >
          <div className={classes.description}>
            <div className={classes['description__item--one']}></div>
            <div className={classes['description__item--two']}>
              <h1 className={classes['description__title']}>
                THE LEADER IN CARING YOU...
              </h1>
              <p className={classes['description__text']}>
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Quos
                maiores possimus rerum delectus magni cupiditate nostrum, ipsa,
                quasi odio facere fuga ea dicta cum debitis in ducimus
                doloremque quod magnam.
              </p>
            </div>
          </div>
        </section>
        <section
          className={`${classes.container} ${classes['container--section2']}`}
        >
          <h1 className={classes['section2__title']}>DISCOVER </h1>
          <a className={classes['section2__title--underline']}>MORE</a>
        </section>
        <section className={classes.galeria}>
          <div className={classes['galeria__images']}>
            {Object.entries(images).map(([key, value], index) => (
              <img
                className={`${classes['galeria__image']} ${
                  classes[`galeria__image--${key}`]
                } ${
                  index === currentImageIndex
                    ? classes['galeria__image--fade']
                    : ''
                }`}
                key={key}
                src={value}
                alt={key}
                loading="lazy"
              />
            ))}
          </div>
        </section>
        <section
          className={`${classes.container} ${classes['container--section3']}`}
        >
          <h1 className={classes['section3__title']}>MEET OUR DOCTORS</h1>
        </section>
        <section
          className={`${classes.container} ${classes['container--section4']}`}
        >
          <div className={classes['section4--carousel']}>
            {doctors.map((doctor: Doctor) => (
              <div className={classes['carousel__card']} key={doctor.id}>
                <div className={classes['carousel__picture']}>
                  <img
                    src={
                      doctor.userId?.image
                        ? doctor.userId?.image
                        : 'https://hips.hearstapps.com/hmg-prod/images/portrait-of-a-happy-young-doctor-in-his-clinic-royalty-free-image-1661432441.jpg?crop=0.66698xw:1xh;center,top&resize=1200:*'
                    }
                    alt={`Dr ${doctor.firstName} ${doctor.lastName}`}
                  />
                </div>
                <div className={classes['carousel__info']}>
                  <h1
                    className={classes['carousel__doctorName']}
                  >{`Dr ${doctor.firstName} ${doctor.lastName}`}</h1>
                  <h1 className={classes['carousel__doctorDepartment']}>
                    {doctor.departmentId.departmentMain}
                  </h1>
                </div>
              </div>
            ))}
          </div>
        </section>
      </div>
    </div>
  );
}
