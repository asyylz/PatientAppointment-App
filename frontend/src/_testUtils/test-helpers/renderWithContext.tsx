import React, { PropsWithChildren } from 'react';
import { render } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { setupStore } from '../../store/index';
import type { AppStore, RootState } from '../../store/index';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  reducer?: any;
  preloadedState?: Partial<RootState>;
  store?: AppStore;
}

export function renderWithProviders(
  ui: React.ReactElement,
  extendedRenderOptions: ExtendedRenderOptions = {}
) {
  const { preloadedState, reducer } = extendedRenderOptions;
  // Ensure that the store is being correctly initialized with preloadedState
  const store = setupStore(reducer, preloadedState);
  //console.log(store.getState()); // Verify the state of the store

  const Wrapper = ({ children }: PropsWithChildren) => (
    <Provider store={store}>{children}</Provider>
  );

  // Return an object with the store and all of RTL's query functions
  return {
    store,
    ...render(ui, { wrapper: Wrapper, ...extendedRenderOptions }),
  };
}
