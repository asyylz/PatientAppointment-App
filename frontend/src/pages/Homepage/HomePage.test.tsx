import HomePage from './HomePage.tsx';
import '@testing-library/jest-dom';
import { act, screen } from '@testing-library/react';
import { fetchDoctors } from '../../store/doctors-slice/doctors-slice';
import { renderWithProviders } from '../../_testUtils/test-helpers/renderWithContext';
import { AppStore } from '../../store/index';
import { combineReducers } from 'redux';
import currentUserReducer from '../../store/currentUser-slice/currentUser-slice';
import doctorsReducer from '../../store/doctors-slice/doctors-slice';
// Mock useEffect globally
jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useEffect: jest.fn(),
}));

const reducers = combineReducers({
  currentUser: currentUserReducer,
  doctors: doctorsReducer,
  // other reducers...
});

const fullfilledAction = {
  type: 'doctors/fetch/fulfilled',
  payload: [
    {
      _id: '665f0f2959c971659af920e5',
      gender: 'F',
      image: '/img/doctors/drmathai_small.jpg',
      phone: '647-722-2370',
      address: {
        street: '390 Steeles Avenue West',
        city: 'Vaughan',
        prov: 'ON',
        postal: 'L4J',
        _id: '665f0f2959c971659af920e7',
      },
      firstName: 'Aytekin',
      lastName: 'Yaliz',
      departmentId: {
        _id: '6660fbd12e3c00fe18cfceec',
        departmentMain: 'Allergy & Immunology',
      },
      userId: {
        _id: '666c1f10ad015e6df17a98e4',
        image:
          'https://patient-appointment-system.s3.eu-west-2.amazonaws.com/1721905296892-doctor.jpg',
      },
      reviews: [],
    },
  ],
};

describe('HomePage snapshot', () => {
  let store: AppStore;
  let asFragment: () => DocumentFragment;

  beforeEach(() => {
    const result = renderWithProviders(<HomePage />, {
      reducer: reducers,
    });

    store = result.store;
    asFragment = result.asFragment;
    jest.spyOn(store, 'dispatch');
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  /* -------------------------- - ------------------------- */
  it('1--Renders correctly before data is fetched', async () => {
    console.log(store.getState());
    expect(asFragment()).toMatchSnapshot();
    await act(async () => {
      store.dispatch(fullfilledAction);
    });
    expect(asFragment()).toMatchSnapshot();
  });

  /* -------------------------- - ------------------------- */
  it('2--Initializes state with status "idle', () => {
    // Initial state should be 'idle'
    expect(store.getState().doctors.status).toEqual('idle');
    expect(store.getState().doctors).toEqual({
      entities: [],
      status: 'idle',
      error: null,
    });
  });
  /* -------------------------- - ------------------------- */
  it('3--Dispatches fetchEntities action and updates status loading', async () => {
    expect(store.getState().doctors.status).toEqual('idle');

    await act(async () => {
      store.dispatch(fetchDoctors('mockURL'));
    });

    expect(store.getState().doctors.status).toEqual('loading');
    console.log(store.getState().doctors.status);
  });
  /* -------------------------- - ------------------------- */
  it('3--Renders fetched data in the DOM after dispatching action', async () => {
    await act(async () => {
      store.dispatch(fullfilledAction);
    });

    // After dispatching the action, the state should be updated
    expect(store.getState().doctors).toEqual({
      entities: [
        {
          _id: '665f0f2959c971659af920e5',
          gender: 'F',
          image: '/img/doctors/drmathai_small.jpg',
          phone: '647-722-2370',
          address: {
            street: '390 Steeles Avenue West',
            city: 'Vaughan',
            prov: 'ON',
            postal: 'L4J',
            _id: '665f0f2959c971659af920e7',
          },
          firstName: 'Aytekin',
          lastName: 'Yaliz',
          departmentId: {
            _id: '6660fbd12e3c00fe18cfceec',
            departmentMain: 'Allergy & Immunology',
          },
          userId: {
            _id: '666c1f10ad015e6df17a98e4',
            image:
              'https://patient-appointment-system.s3.eu-west-2.amazonaws.com/1721905296892-doctor.jpg',
          },
          reviews: [],
        },
      ],
      status: 'succeeded',
      error: null,
    });
    expect(screen.getByText('Dr Aytekin Yaliz')).toBeInTheDocument();
    expect(screen.getByText('Allergy & Immunology')).toBeInTheDocument();
    expect(screen.getByAltText('Dr Aytekin Yaliz')).toHaveAttribute(
      'src',
      'https://patient-appointment-system.s3.eu-west-2.amazonaws.com/1721905296892-doctor.jpg'
    );
    expect(screen.getByAltText('Dr Aytekin Yaliz')).not.toHaveAttribute(
      'src',
      'https://hips.hearstapps.com/hmg-prod/images/portrait-of-a-happy-young-doctor-in-his-clinic-royalty-free-image-1661432441.jpg?crop=0.66698xw:1xh;center,top&resize=1200:*'
    );
  });
});
