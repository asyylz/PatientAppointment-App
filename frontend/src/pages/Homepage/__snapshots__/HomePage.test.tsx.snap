// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage snapshot 1--Renders correctly before data is fetched 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      class="container container--body"
    >
      <section
        class="container container--section1"
      >
        <div
          class="description"
        >
          <div
            class="description__item--one"
          />
          <div
            class="description__item--two"
          >
            <h1
              class="description__title"
            >
              THE LEADER IN CARING YOU...
            </h1>
            <p
              class="description__text"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Quos maiores possimus rerum delectus magni cupiditate nostrum, ipsa, quasi odio facere fuga ea dicta cum debitis in ducimus doloremque quod magnam.
            </p>
          </div>
        </div>
      </section>
      <section
        class="container container--section2"
      >
        <h1
          class="section2__title"
        >
          DISCOVER 
        </h1>
        <a
          class="section2__title--underline"
        >
          MORE
        </a>
      </section>
      <section
        class="galeria"
      >
        <div
          class="galeria__images"
        >
          <img
            alt="one"
            class="galeria__image galeria__image--one galeria__image--fade"
            data-testid="current-image"
            loading="lazy"
            src="https://images.unsplash.com/photo-1499728603263-13726abce5fd?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8NHx8aGVhbHRofGVufDB8fDB8fHwy"
          />
          <img
            alt="two"
            class="galeria__image galeria__image--two "
            data-testid=""
            loading="lazy"
            src="https://images.unsplash.com/photo-1488228469209-c141f8bcd723?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MzB8fGhlYWx0aHxlbnwwfHwwfHx8Mg%3D%3D"
          />
          <img
            alt="three"
            class="galeria__image galeria__image--three "
            data-testid=""
            loading="lazy"
            src="https://images.unsplash.com/photo-1444312645910-ffa973656eba?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8Nnx8aGVhbHRofGVufDB8fDB8fHwy"
          />
          <img
            alt="four"
            class="galeria__image galeria__image--four "
            data-testid=""
            loading="lazy"
            src="https://images.unsplash.com/photo-1530497610245-94d3c16cda28?q=80&w=1364&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D"
          />
          <img
            alt="five"
            class="galeria__image galeria__image--five "
            data-testid=""
            loading="lazy"
            src="https://images.unsplash.com/photo-1604881991720-f91add269bed?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8Mjc1fHxoZWFsdGh8ZW58MHx8MHx8fDI%3D"
          />
        </div>
      </section>
      <section
        class="container container--section3"
      >
        <h1
          class="section3__title"
        >
          MEET OUR DOCTORS
        </h1>
      </section>
      <section
        class="container container--section4"
      >
        <div
          class="section4--carousel"
        />
      </section>
    </div>
  </div>
</DocumentFragment>
`;

exports[`HomePage snapshot 1--Renders correctly before data is fetched 2`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      class="container container--body"
    >
      <section
        class="container container--section1"
      >
        <div
          class="description"
        >
          <div
            class="description__item--one"
          />
          <div
            class="description__item--two"
          >
            <h1
              class="description__title"
            >
              THE LEADER IN CARING YOU...
            </h1>
            <p
              class="description__text"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Quos maiores possimus rerum delectus magni cupiditate nostrum, ipsa, quasi odio facere fuga ea dicta cum debitis in ducimus doloremque quod magnam.
            </p>
          </div>
        </div>
      </section>
      <section
        class="container container--section2"
      >
        <h1
          class="section2__title"
        >
          DISCOVER 
        </h1>
        <a
          class="section2__title--underline"
        >
          MORE
        </a>
      </section>
      <section
        class="galeria"
      >
        <div
          class="galeria__images"
        >
          <img
            alt="one"
            class="galeria__image galeria__image--one galeria__image--fade"
            data-testid="current-image"
            loading="lazy"
            src="https://images.unsplash.com/photo-1499728603263-13726abce5fd?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8NHx8aGVhbHRofGVufDB8fDB8fHwy"
          />
          <img
            alt="two"
            class="galeria__image galeria__image--two "
            data-testid=""
            loading="lazy"
            src="https://images.unsplash.com/photo-1488228469209-c141f8bcd723?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MzB8fGhlYWx0aHxlbnwwfHwwfHx8Mg%3D%3D"
          />
          <img
            alt="three"
            class="galeria__image galeria__image--three "
            data-testid=""
            loading="lazy"
            src="https://images.unsplash.com/photo-1444312645910-ffa973656eba?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8Nnx8aGVhbHRofGVufDB8fDB8fHwy"
          />
          <img
            alt="four"
            class="galeria__image galeria__image--four "
            data-testid=""
            loading="lazy"
            src="https://images.unsplash.com/photo-1530497610245-94d3c16cda28?q=80&w=1364&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D"
          />
          <img
            alt="five"
            class="galeria__image galeria__image--five "
            data-testid=""
            loading="lazy"
            src="https://images.unsplash.com/photo-1604881991720-f91add269bed?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8Mjc1fHxoZWFsdGh8ZW58MHx8MHx8fDI%3D"
          />
        </div>
      </section>
      <section
        class="container container--section3"
      >
        <h1
          class="section3__title"
        >
          MEET OUR DOCTORS
        </h1>
      </section>
      <section
        class="container container--section4"
      >
        <div
          class="section4--carousel"
        >
          <div
            class="carousel__card"
          >
            <div
              class="carousel__picture"
            >
              <img
                alt="Dr Aytekin Yaliz"
                src="https://patient-appointment-system.s3.eu-west-2.amazonaws.com/1721905296892-doctor.jpg"
              />
            </div>
            <div
              class="carousel__info"
            >
              <h1
                class="carousel__doctorName"
              >
                Dr Aytekin Yaliz
              </h1>
              <h1
                class="carousel__doctorDepartment"
              >
                Allergy & Immunology
              </h1>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
</DocumentFragment>
`;
